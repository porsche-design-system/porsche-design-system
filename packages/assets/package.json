{
  "name": "@porsche-design-system/assets",
  "version": "4.0.0",
  "scope": "@porsche-design-system",
  "main": "dist/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "sideEffects": false,
  "author": "Dr. Ing. h.c. F. Porsche AG",
  "contributors": [
    "Stephan Schröter <stephan.schroeter@mhp.com>",
    "Oliver Hook <oliver.hook@mhp.com>",
    "Marcel Bertram <marcel.bertram@porsche.de>",
    "Matthias Hägele <matthias.haegele@mhp.com>",
    "Olivia Pala <olivia.pala@udg.de>",
    "Anna Krohmer <anna.krohmer@udg.de>",
    "Melanie Kolar <melanie.kolar@udg.de>",
    "Bernd Hacker <bernd.hacker@mhp.com>",
    "Anton Schmidt <anton.schmidt@mhp.com>"
  ],
  "license": "SEE LICENSE IN https://designsystem.porsche.com",
  "description": "Porsche Design System Assets",
  "keywords": [
    "porsche",
    "assets"
  ],
  "homepage": "https://designsystem.porsche.com",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/porscheui/porsche-design-system.git"
  },
  "bugs": {
    "url": "https://github.com/porscheui/porsche-ui-contribution/issues"
  },
  "scripts": {
    "start": "yarn build && node ./bin/serve-cdn.js",
    "build": "yarn clean:build && yarn build:bin && yarn build:manifest && yarn build:copyAssets && yarn build:preparePackage",
    "build:bin": "tsc --build tsconfig.node.json && chmod +x bin/serve-cdn.js",
    "build:copyAssets": "COMPONENTS_ONLY=${COMPONENTS_ONLY:-0} ts-node --project tsconfig.node.json scripts/copyAssets.ts",
    "build:manifest": "rollup -c rollup.config.js",
    "build:preparePackage": "ts-node --project tsconfig.node.json scripts/preparePackage.ts",
    "buildCopyComponentsOnly": "rm -rf ./cdn/components && COMPONENTS_ONLY=1 yarn build:copyAssets",
    "deployFrontify": "ts-node --project tsconfig.node.json scripts/deployFrontify.ts",
    "clean": "yarn clean:node_modules && yarn clean:build",
    "clean:node_modules": "rm -rf ./node_modules && rm -rf ./yarn-error.log",
    "clean:build": "rm -rf ./dist && rm -rf ./cdn && rm -rf ./bin"
  },
  "dependencies": {
    "@porsche-design-system/email": "0.0.0",
    "@porsche-design-system/fonts": "0.0.0",
    "@porsche-design-system/icons": "0.0.0",
    "@porsche-design-system/marque": "0.0.0",
    "@porsche-design-system/meta-icons": "0.0.0",
    "@porsche-design-system/utilities-v2": "0.0.0"
  },
  "devDependencies": {
    "@rollup/plugin-node-resolve": "^13.2.1",
    "@rollup/plugin-typescript": "^8.3.2",
    "@types/http-server": "^0.12.1",
    "@types/node-fetch": "^2.6.1",
    "http-server": "^14.1.0",
    "node-fetch": "^2.6.7",
    "rollup": "^2.71.1",
    "ts-node": "^10.7.0",
    "typescript": "~4.6.4"
  },
  "files": [
    "dist/"
  ],
  "bin": {
    "serve-cdn": "./bin/serve-cdn.js"
  }
}
