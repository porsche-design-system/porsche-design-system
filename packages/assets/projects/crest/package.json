{
  "name": "@porsche-design-system/crest",
  "version": "0.0.0",
  "private": true,
  "scope": "@porsche-design-system",
  "main": "dist/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "author": "Dr. Ing. h.c. F. Porsche AG",
  "license": "UNLICENSED",
  "description": "Porsche Crest",
  "keywords": [
    "porsche",
    "crest"
  ],
  "homepage": "https://designsystem.porsche.com",
  "scripts": {
    "build": "yarn clean:build && yarn build:crest && yarn build:lib",
    "build:crest": "tsx scripts/build.ts",
    "build:lib": "yarn build:lib:cjs && yarn build:lib:esm",
    "build:lib:cjs": "tsc --build tsconfig.lib.cjs.json",
    "build:lib:esm": "tsc --build tsconfig.lib.esm.json",
    "clean:build": "rm -rf ./index.ts && rm -rf ./dist"
  },
  "devDependencies": {
    "change-case": "^5.4.4",
    "globby": "^14.0.1",
    "tsx": "^4.7.2",
    "typescript": "~5.4.4"
  },
  "files": [
    "dist/"
  ],
  "volta": {
    "extends": "../../../../package.json"
  }
}
