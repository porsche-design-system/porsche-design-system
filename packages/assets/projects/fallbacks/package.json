{
  "name": "@porsche-design-system/fallbacks",
  "version": "0.0.0",
  "private": true,
  "license": "UNLICENSED",
  "scope": "@porsche-design-system",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.mts",
        "default": "./dist/index.mjs"
      },
      "require": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    }
  },
  "types": "./dist/index.d.ts",
  "files": ["./dist/"],
  "author": "Dr. Ing. h.c. F. Porsche AG",
  "homepage": "https://designsystem.porsche.com",
  "scripts": {
    "start": "yarn build && http-server ./www -p 4201",
    "start-app": "yarn start",
    "build": "yarn clean:build && yarn build:fallbacks && yarn build:lib && yarn build:www",
    "build:fallbacks": "yarn build:fallbacks:generate-overlay && yarn build:fallbacks:tmp && yarn build:fallbacks:cdn+loader",
    "build:fallbacks:generate-overlay": "tsx scripts/generate-overlay.ts",
    "build:fallbacks:tmp": "rollup -c rollup.config.js --bundleConfigAsCjs",
    "build:fallbacks:cdn+loader": "tsx scripts/generate-cdn-loader.ts",
    "build:lib": "tsup index.ts --format esm,cjs --dts",
    "build:www": "rm -rf ./www && mkdir -p ./www && cp -r ./public/. ./www && cp -r ./dist/tmp/. ./www/.",
    "test:unit": "vitest --config tests/unit/config/vitest.config.ts",
    "test:vrt": "playwright test --config=tests/vrt/config/playwright.config.ts",
    "clean:build": "rm -rf ./index.ts && rm -rf ./dist && rm -rf ./cdn && rm -rf ./tmp && rm -rf ./www"
  },
  "devDependencies": {
    "@porsche-design-system/shared": "0.0.0",
    "@types/html-minifier": "^4.0.5",
    "html-minifier": "^4.0.0",
    "http-server": "^14.1.1",
    "tslib": "^2.8.1"
  },
  "peerDependencies": {
    "@playwright/test": "*",
    "change-case": "*",
    "tsup": "*",
    "tsx": "*",
    "typescript": "*"
  },
  "volta": {
    "extends": "../../../../package.json"
  }
}
