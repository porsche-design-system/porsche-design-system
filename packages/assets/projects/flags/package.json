{
  "name": "@porsche-design-system/flags",
  "version": "0.0.0",
  "private": true,
  "license": "UNLICENSED",
  "scope": "@porsche-design-system",
  "type": "commonjs",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.mts",
        "default": "./dist/index.mjs"
      },
      "require": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    }
  },
  "types": "./dist/index.d.ts",
  "files": [
    "./dist/"
  ],
  "author": "Dr. Ing. h.c. F. Porsche AG",
  "homepage": "https://designsystem.porsche.com",
  "scripts": {
    "lint": "publint",
    "build": "yarn clean:build && yarn build:flags && yarn build:lib",
    "build:flags": "tsx scripts/build.ts",
    "build:lib": "tsup index.ts --format esm,cjs --dts",
    "test:unit": "vitest --config tests/unit/config/vitest.config.ts",
    "clean:build": "rm -rf ./index.ts && rm -rf ./dist"
  },
  "devDependencies": {
    "@porsche-design-system/shared": "0.0.0"
  },
  "peerDependencies": {
    "change-case": "*",
    "globby": "*",
    "prettier": "*",
    "publint": "*",
    "svgo": "*",
    "tsup": "*",
    "tsx": "*",
    "typescript": "*",
    "vitest": "*",
    "gzip-size": "*"
  },
  "volta": {
    "extends": "../../../../package.json"
  }
}
