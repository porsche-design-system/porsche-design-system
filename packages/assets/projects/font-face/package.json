{
  "name": "@porsche-design-system/font-face",
  "version": "0.0.0",
  "private": true,
  "license": "UNLICENSED",
  "scope": "@porsche-design-system",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.mts",
        "default": "./dist/index.mjs"
      },
      "require": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    }
  },
  "types": "./dist/index.d.ts",
  "files": ["./dist/"],
  "author": "Dr. Ing. h.c. F. Porsche AG",
  "homepage": "https://designsystem.porsche.com",
  "scripts": {
    "build": "yarn clean:build && yarn build:font-face && yarn build:lib",
    "build:font-face": "tsx scripts/buildFontFace.ts",
    "build:lib": "tsup index.ts --format esm,cjs --dts",
    "test:unit": "vitest --config tests/unit/config/vitest.config.ts",
    "clean:build": "rm -rf ./index.ts && rm -rf ./dist"
  },
  "devDependencies": {
    "@porsche-design-system/fonts": "0.0.0",
    "@porsche-design-system/shared": "0.0.0"
  },
  "peerDependencies": {
    "change-case": "*",
    "jss": "*",
    "tsup": "*",
    "tsx": "*",
    "typescript": "*"
  },
  "volta": {
    "extends": "../../../../package.json"
  }
}
