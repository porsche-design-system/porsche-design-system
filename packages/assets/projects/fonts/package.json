{
  "name": "@porsche-design-system/fonts",
  "version": "0.0.0",
  "private": true,
  "scope": "@porsche-design-system",
  "main": "dist/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "author": "Dr. Ing. h.c. F. Porsche AG",
  "license": "UNLICENSED",
  "description": "Porsche Fonts",
  "keywords": [
    "porsche",
    "fonts"
  ],
  "homepage": "https://designsystem.porsche.com",
  "scripts": {
    "build": "yarn clean:build && yarn build:fonts && yarn build:lib",
    "build:fonts": "tsx scripts/build.ts",
    "build:lib": "yarn build:lib:cjs && yarn build:lib:esm",
    "build:lib:cjs": "tsc --build tsconfig.lib.cjs.json",
    "build:lib:esm": "tsc --build tsconfig.lib.esm.json",
    "clean:build": "rm -rf ./index.ts && rm -rf ./dist"
  },
  "devDependencies": {
    "change-case": "^5.4.4",
    "globby": "^14.0.1",
    "tsx": "^4.7.2",
    "typescript": "~5.4.4"
  },
  "files": [
    "dist/"
  ],
  "volta": {
    "extends": "../../../../package.json"
  }
}
