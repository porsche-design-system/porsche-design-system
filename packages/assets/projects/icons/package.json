{
  "name": "@porsche-design-system/icons",
  "version": "0.0.0",
  "private": true,
  "scope": "@porsche-design-system",
  "main": "dist/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "author": "Dr. Ing. h.c. F. Porsche AG",
  "license": "UNLICENSED",
  "description": "Porsche Icons",
  "keywords": [
    "porsche",
    "icons"
  ],
  "homepage": "https://icons.porsche.com",
  "scripts": {
    "build": "yarn clean:build && yarn build:icons && yarn build:lib",
    "build:icons": "ts-node scripts/build.ts",
    "build:lib": "yarn build:lib:cjs && yarn build:lib:esm",
    "build:lib:cjs": "tsc --build tsconfig.lib.cjs.json",
    "build:lib:esm": "tsc --build tsconfig.lib.esm.json",
    "test:unit": "jest --colors -c tests/unit/config/jest.config.js",
    "clean": "yarn clean:node_modules && yarn clean:build",
    "clean:node_modules": "rm -rf ./node_modules && rm -rf ./yarn-error.log",
    "clean:build": "rm -rf ./index.ts && rm -rf ./dist"
  },
  "devDependencies": {
    "@porsche-design-system/shared": "0.0.0",
    "@types/js-yaml": "^4.0.9",
    "change-case": "^4.1.2",
    "globby": "^11.0.4",
    "js-yaml": "^4.0.0",
    "svgo": "^3.1.0",
    "ts-node": "^10.9.2",
    "typescript": "~5.3.3"
  },
  "files": [
    "dist/"
  ],
  "volta": {
    "extends": "../../../../package.json"
  }
}
