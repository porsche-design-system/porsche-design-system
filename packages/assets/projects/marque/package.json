{
  "name": "@porsche-design-system/marque",
  "version": "0.0.0",
  "private": true,
  "license": "UNLICENSED",
  "scope": "@porsche-design-system",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.mts",
        "default": "./dist/index.mjs"
      },
      "require": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    }
  },
  "types": "./dist/index.d.ts",
  "files": [
    "./dist/"
  ],
  "author": "Dr. Ing. h.c. F. Porsche AG",
  "homepage": "https://designsystem.porsche.com",
  "scripts": {
    "build": "yarn clean:build && yarn build:marque && yarn build:lib",
    "build:marque": "tsx scripts/build.ts",
    "build:lib": "tsup index.ts --format esm,cjs --dts",
    "clean:build": "rm -rf ./index.ts && rm -rf ./dist"
  },
  "devDependencies": {
    "@porsche-design-system/shared": "0.0.0"
  },
  "peerDependencies": {
    "change-case": "*",
    "globby": "*",
    "tsup": "*",
    "tsx": "*",
    "typescript": "*"
  },
  "volta": {
    "extends": "../../../../package.json"
  }
}
