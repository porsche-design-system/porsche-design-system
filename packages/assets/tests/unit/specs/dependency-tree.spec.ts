// @ts-ignore
import madge from 'madge';
import * as path from 'path';
import { main, module } from '../../../dist/package.json';

const packageDir = path.resolve(__dirname, '../../../dist');

it('should have no regression for esm dependency tree', async () => {
  const tree = await madge(`${packageDir}/${module}`);

  // visualize result
  // await tree.image('result.svg');

  expect(tree.obj()).toMatchSnapshot();
}, 30000);

it('should have no regression for cjs dependency tree', async () => {
  const tree = await madge(`${packageDir}/${main}`);

  // visualize result
  // await tree.image('result.svg')

  expect(tree.obj()).toMatchSnapshot();
});
