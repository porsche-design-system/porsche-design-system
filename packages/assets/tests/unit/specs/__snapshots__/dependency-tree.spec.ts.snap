// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have no regression for cjs dependency tree 1`] = `
{
  "../../crest/dist/esm/index.cjs": [],
  "../../fallbacks/dist/esm/index.cjs": [],
  "../../fonts/dist/esm/index.cjs": [],
  "../../icons/dist/esm/index.cjs": [],
  "../../marque/dist/esm/index.cjs": [],
  "../../meta-icons/dist/esm/index.cjs": [],
  "../../model-signatures/dist/esm/index.cjs": [],
  "../../utilities/projects/styles/dist/esm/index.cjs": [],
  "index.cjs": [
    "../../crest/dist/esm/index.cjs",
    "../../fallbacks/dist/esm/index.cjs",
    "../../fonts/dist/esm/index.cjs",
    "../../icons/dist/esm/index.cjs",
    "../../marque/dist/esm/index.cjs",
    "../../meta-icons/dist/esm/index.cjs",
    "../../model-signatures/dist/esm/index.cjs",
    "../../utilities/projects/styles/dist/esm/index.cjs",
  ],
}
`;

exports[`should have no regression for esm dependency tree 1`] = `
{
  "../../crest/dist/esm/index.mjs": [],
  "../../fallbacks/dist/esm/index.mjs": [],
  "../../fonts/dist/esm/index.mjs": [],
  "../../icons/dist/esm/index.mjs": [],
  "../../marque/dist/esm/index.mjs": [],
  "../../meta-icons/dist/esm/index.mjs": [],
  "../../model-signatures/dist/esm/index.mjs": [],
  "../../utilities/projects/styles/dist/esm/index.mjs": [],
  "index.mjs": [
    "../../crest/dist/esm/index.mjs",
    "../../fallbacks/dist/esm/index.mjs",
    "../../fonts/dist/esm/index.mjs",
    "../../icons/dist/esm/index.mjs",
    "../../marque/dist/esm/index.mjs",
    "../../meta-icons/dist/esm/index.mjs",
    "../../model-signatures/dist/esm/index.mjs",
    "../../utilities/projects/styles/dist/esm/index.mjs",
  ],
}
`;
